tests/core/policies/test_rule_policy.py

```diff
@@ -922,6 +920,113 @@ async def test_predict_form_action_if_multiple_turns():
     assert_predicted_action(prediction, domain, form_name, is_no_user_prediction=True)
 
 
+async def test_predict_slot_initial_value_not_required_in_rule():
+    domain = Domain.from_yaml(
+        """
+intents:
+- i1
+actions:
+- action1
+slots:
+  s_cat1:
+    type: categorical
+    values:
+      - v1
+      - v2
+    initial_value: v1
+"""
+    )
+
+    rule = DialogueStateTracker.from_events(
+        "slot rule",
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": "i1"}),
+            ActionExecuted("action1"),
+            SlotSet("s_cat1", "v2"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+        domain=domain,
+        slots=domain.slots,
+    )
+    rule.is_rule_tracker = True
+
+    policy = RulePolicy()
+    policy.train([rule], domain, RegexInterpreter())
+
+    form_conversation = DialogueStateTracker.from_events(
+        "slot rule test",
+        evts=[
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": "i1"}),
+            ActionExecuted("action1"),
+            SlotSet("s_cat1", "v2"),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": "i1"}),
+        ],
+        domain=domain,
+        slots=domain.slots,
+    )
+
+    prediction = policy.predict_action_probabilities(
+        form_conversation, domain, RegexInterpreter()
+    )
+    assert_predicted_action(prediction, domain, "action1")
+
+
+async def test_predict_slot_with_initial_slot_matches_rule():
+    domain = Domain.from_yaml(
+        """
+intents:
+- i1
+actions:
+- action1
+slots:
+  s_cat1:
+    type: categorical
+    values:
+      - v1
+      - v2
+    initial_value: v1
+"""
+    )
+
+    rule = DialogueStateTracker.from_events(
+        "slot rule",
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            SlotSet("s_cat1", "v1"),
+            UserUttered(intent={"name": "i1"}),
+            ActionExecuted(ACTION_LISTEN_NAME),
```

Comment:
I think this event should be removed.

Reply:
why do you think it should be removed?

Reply:
Afaik `action_listen` only ever goes after some action, not after intents. It might actually be that somewhere down the line this `action_listen` gets removed/ignored as it has no real meaning/role here 🤔

Reply:
ah yes it's after the `UserUttered`! Good spot, I didn't see that.